cache.go

```diff
@@ -205,25 +212,27 @@ func NewCache(config *Config) (*Cache, error) {
 	//       goroutines we have running cache.processItems(), so 1 should
 	//       usually be sufficient
 	go cache.processItems()
+
 	return cache, nil
 }
 
-func (c *Cache) Wait() {
+func (c *Cache[K, V]) Wait() {
 	if c == nil || c.isClosed {
 		return
 	}
 	wg := &sync.WaitGroup{}
 	wg.Add(1)
-	c.setBuf <- &Item{wg: wg}
+	c.setBuf <- &Item[V]{wg: wg}
 	wg.Wait()
 }
 
 // Get returns the value (if any) and a boolean representing whether the
 // value was found or not. The value can be nil and the boolean can be true at
 // the same time.
-func (c *Cache) Get(key interface{}) (interface{}, bool) {
-	if c == nil || c.isClosed || key == nil {
-		return nil, false
+func (c *Cache[K, V]) Get(key K) (V, bool) {
+
+	if c == nil || c.isClosed || &key == nil {
```

Comment:
pointer to key can never be nil. That is why I am wondering whether we should take a pointer to K (*K) as the arg to Get function